javaslang/src/main/java/javaslang/collection/CharSeq.java

```diff
@@ -2108,18 +2088,14 @@ public CharSeq replace(CharSequence target, CharSequence replacement) {
      *
      * @param regex the delimiting regular expression
      * @param limit the result threshold, as described above
-     * @return  the Seq of strings computed by splitting this string
+     * @return the Seq of strings computed by splitting this string
      * around matches of the given regular expression
      * @throws PatternSyntaxException if the regular expression's syntax is invalid
      * @see java.util.regex.Pattern
      */
     public Seq<CharSeq> split(String regex, int limit) {
-        final String[] javaStrings = back.split(regex, limit);
-        final CharSeq[] strings = new CharSeq[javaStrings.length];
-        for (int i = 0; i < strings.length; i++) {
-            strings[i] = of(javaStrings[i]);
-        }
-        return Vector.of(strings);
+        Array<String> split = Array.wrap(back.split(regex, limit));
```

Comment:
Will add local finals all over the code in the cleanup commit, if that's how you prefer it :)
I think Java is going in the 'implicitly final' direction though, but I can see how this will add value also.